presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/jmx/StatisticsAwareConnectionFactory.java

```diff
@@ -42,6 +44,7 @@ public Connection openConnection(JdbcIdentity identity)
         return openConnection.wrap(() -> delegate.openConnection(identity));
     }
 
+    @PreDestroy
```

Comment:
Good, but what if `StatisticsAwareConnectionFactory` is not used (e.g. we make stats optional one day)? Maybe add this to other `ConnectionFactory` classes?

Reply:
There is only `DriverConnectorFactory` which does not allocate anything